tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,3176 @@
+    const Solution& solution, const size_t grid_size_each_dimension,
+    const std::array<double, 3>& lower_bound,
+    const std::array<double, 3>& upper_bound, std::mt19937 generator) noexcept {
+  // Initialize random number generation
+  std::uniform_real_distribution<> pdist(0.01, 1.);
+  std::uniform_int_distribution<> idist(2, 7);
+
+  // Setup grid
+  const size_t SpatialDim = 3;
+  Mesh<SpatialDim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                        Spectral::Quadrature::GaussLobatto};
+
+  const auto coord_map =
+      domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(Affine3D{
+          Affine{-1., 1., lower_bound[0], upper_bound[0]},
+          Affine{-1., 1., lower_bound[1], upper_bound[1]},
+          Affine{-1., 1., lower_bound[2], upper_bound[2]},
+      });
+
+  // Setup coordinates
+  const auto x_logical = logical_coordinates(mesh);
+  const auto x = coord_map(x_logical);
+  // Arbitrary time for time-independent solution.
+  const double t = pdist(generator) * 0.4;
+
+  // Evaluate analytic solution
+  const auto vars =
+      solution.variables(x, t, typename Solution::template tags<DataVector>{});
+  const auto& lapse = get<gr::Tags::Lapse<>>(vars);
+  const auto& shift = get<gr::Tags::Shift<SpatialDim>>(vars);
+  const auto& spatial_metric = get<gr::Tags::SpatialMetric<SpatialDim>>(vars);
+
+  // Get ingredients
+  const auto& spacetime_metric =
+      gr::spacetime_metric(lapse, shift, spatial_metric);
+  const auto& spacetime_unit_normal_one_form =
+      gr::spacetime_normal_one_form<SpatialDim, Frame::Inertial, DataVector>(
+          lapse);
+  const auto& det_spatial_metric =
+      determinant_and_inverse(spatial_metric).first;
+  Scalar<DataVector> sqrt_det_spatial_metric(sqrt(get(det_spatial_metric)));
+
+  // Initialize settings
+  const double amp_coef_L1 = pdist(generator);
+  const int exp_L1 = idist(generator);
+  // roll on function parameters for lapse / shift terms
+  const double t_start_L1 = pdist(generator) * 0.1;
+  const double sigma_t_L1 = pdist(generator) * 0.2;
+  const double r_max = pdist(generator) * 0.7;
+  const double roll_on_L1 =
+      GeneralizedHarmonic::DampedHarmonicGauge_detail::roll_on_function(
+          t, t_start_L1, sigma_t_L1);
+  const auto& weight =
+      GeneralizedHarmonic::DampedHarmonicGauge_detail::weight_function<
+          SpatialDim, Frame::Inertial, DataVector>(x, r_max);
+
+  const auto gauge_h_init = make_with_value<typename db::item_type<
+      GeneralizedHarmonic::Tags::InitialGaugeH<SpatialDim, Frame::Inertial>>>(
+      x, 0.);
+  typename db::item_type<
+      GeneralizedHarmonic::Tags::GaugeH<SpatialDim, Frame::Inertial>>
+      gauge_h{};
+  GeneralizedHarmonic::damped_harmonic_h<SpatialDim, Frame::Inertial>(
+      make_not_null(&gauge_h), gauge_h_init, lapse, shift,
+      sqrt_det_spatial_metric, spacetime_metric, t, x, amp_coef_L1, 0., 0.,
+      exp_L1, 0, 0, /* exponents */
```

Comment:
Please use C++ style comments where the comment is at the end of the line

Reply:
done